book/src/creating_templates.md

```diff
@@ -108,3 +108,42 @@ recognized:
   #[template(path = "hello.html", config = "config.toml")]
   struct HelloTemplate<'a> { ... }
   ```
+
+## Documentation as template code
+
+As an alternative to supplying the code template code in an external file (as `path` argument),
+or as a string (as `source` argument), you can also enable the `"code-in-doc"` feature.
+With this feature, you can specify the template code directly in the documentation
+of the template `struct`. Simply add a ```` ```rinja ```` code block to your documentation:
+
+```rust
+/// Here you can put our usual comments.
+///
+/// ```rinja
+/// <div>{{ lines|linebreaksbr }}</div>
+/// ```
+///
+/// Any usual docs, including tests can be put in here, too:
+///
+/// ```rust
+/// assert_eq!(
+///     Example { lines: "a\nb\nc" }.to_string(),
+///     "<div>a<br/>b<br/>c</div>"
+/// );
+/// ```
+///
+/// All comments are still optional, though.
+#[derive(Template)]
+#[template(ext = "html")]
```

Comment:
I thought about it and I think we should only parse documentation if we have an attribute like `#[template(in_doc = true)]` or something like that. I see two reasons why we should do:
 1. For users, it's not magical anymore when they read source code to know where the template comes from.
 2. For `rinja`, it means we'll only parse doc comments if and only if the attribute is set to `true` (so impact on performance reduced).
 
 What do you think?

Reply:
Yes, makes sense! Added.